test/unit/lodash.spec.js

```diff
@@ -0,0 +1,690 @@
+                expect(element2).to.eql('string');
+                expect(params.length).to.eql(2);
+            });
+            it('should contain array of objects', function () {
+                var array = params[0];
+                for (var i = 0; i < array.length; i++) {
+                    expect(Object.keys(array[i]).length).to.eql(2);
+                }
+
+                expect(array.length).to.eql(3);
+                array.forEach(function (n) {
+                    expect(_.has(n, 'ALIVE')).to.eql(true);
+                });
+            });
+            it('should return true if all elements pass the predicate "ALIVE" check', function () {
+                expect(_.every.apply(_, datasets.every())).to.eql(true);
+            });
+        });
+        describe('every2 function', function () {
+            var params = datasets.every2();
+            var array = params[0];
+            var arrLenght = array.length;
+
+            it('should match types of passing elements', function () {
+                var element1 = params[0] instanceof Array;
+                var element2 = params[1] instanceof Function;
+
+                expect(element1).to.eql(true);
+                expect(element2).to.eql(true);
+                expect(params.length).to.eql(2);
+            });
+            it('should contain array of different type elements which are Booleanly true', function () {
+                var types = _.map(array, function (i) {
+                    array = typeof i;
+
+                    return array;
+                });
+                var result = _.uniq(types);
+
+                expect(result.length).to.eql(arrLenght);
+            });
+
+            it('should return true if all elements are Booleanly true', function () {
+                expect(arrLenght).to.eql(4);
+                expect(_.every.apply(_, datasets.every2())).to.eql(true);
+            });
+        });
+    });
+
+    describe('filter', function () {
+        describe('filter function', function () {
+            var params = datasets.filter();
+
+            it('should match types of passing elements', function () {
+                var element1 = params[0] instanceof Array;
+                var element2 = params[1] instanceof Function;
+
+                expect(element1).to.eql(true);
+                expect(element2).to.eql(true);
+                expect(params.length).to.eql(2);
+            });
+            it('should contain array of 5 object with length of 3, each', function () {
+                var array = params[0];
+
+                expect(array.length).to.eql(5);
+                array.forEach(function (obj) {
+                    if (Object.keys(obj).length < 3 || Object.keys(obj).length > 3) {
+                        expect(Object.keys(obj).length).to.eql(3);
+                    }
+                });
+            });
+            it('should invoke passed function as predicate matching age lower than 40', function () {
```

Comment:
I was able to hack this case just by using pseudos & other properties of the object. Consider checking iteratee function itself, maybe without actual underscore tests?